opslevel/resource_opslevel_team.go

```diff
@@ -108,63 +118,81 @@ func reconcileTeamAliases(d *schema.ResourceData, team *opslevel.Team, client *o
 	return nil
 }
 
-func collectMembersFromTeam(team *opslevel.Team) []string {
-	members := []string{}
+func collectMembersFromTeam(team *opslevel.Team) []opslevel.TeamMembershipUserInput {
+	members := []opslevel.TeamMembershipUserInput{}
 
 	for _, user := range team.Members.Nodes {
-		members = append(members, user.Email)
+		member := opslevel.TeamMembershipUserInput{
+			User: opslevel.UserIdentifierInput{
+				Email: user.Email,
+			},
+			Role: string(user.Role),
+		}
+		members = append(members, member)
 	}
 	return members
 }
 
+func memberInArray(member opslevel.TeamMembershipUserInput, array []opslevel.TeamMembershipUserInput) bool {
+	for _, m := range array {
+		if m.User.Email == member.User.Email && m.Role == member.Role {
+			return true
+		}
+	}
+	return false
+}
+
 func reconcileTeamMembership(d *schema.ResourceData, team *opslevel.Team, client *opslevel.Client) error {
-	expectedMembers := expandStringArray(d.Get("members").(*schema.Set).List())
+	expectedMembers := []opslevel.TeamMembershipUserInput{}
 	existingMembers := collectMembersFromTeam(team)
 
-	membersToRemove := []string{}
-	membersToAdd := []string{}
+	if members, ok := d.GetOk("member"); ok {
+		membersInput := members.([]interface{})
+
+		for _, m := range membersInput {
+			memberInput := m.(map[string]interface{})
+			member := opslevel.TeamMembershipUserInput{
+				User: opslevel.UserIdentifierInput{
+					Email: memberInput["email"].(string),
+				},
+				Role: memberInput["role"].(string),
+			}
+			expectedMembers = append(expectedMembers, member)
+		}
+	}
+
+	membersToRemove := []opslevel.TeamMembershipUserInput{}
+	membersToAdd := []opslevel.TeamMembershipUserInput{}
 
 	for _, existingMember := range existingMembers {
-		if stringInArray(existingMember, expectedMembers) {
+		if memberInArray(existingMember, expectedMembers) {
 			continue
 		}
 
 		membersToRemove = append(membersToRemove, existingMember)
 	}
 
 	for _, expectedMember := range expectedMembers {
-
-		if stringInArray(expectedMember, existingMembers) {
+		if memberInArray(expectedMember, existingMembers) {
 			continue
 		}
 		membersToAdd = append(membersToAdd, expectedMember)
 	}
 
 	if len(membersToAdd) != 0 {
-		_, err := client.AddMembers(&team.TeamId, membersToAdd)
+		_, err := client.AddMemberships(&team.TeamId, membersToAdd...)
 		if err != nil {
 			return err
 		}
 	}
 
 	if len(membersToRemove) != 0 {
-		_, err := client.RemoveMembers(&team.TeamId, membersToRemove)
+		_, err := client.RemoveMemberships(&team.TeamId, membersToRemove...)
 		if err != nil {
 			return err
 		}
 	}
-	return nil
-}
 
-func validateMembershipState(d *schema.ResourceData) error {
```

Comment:
This restriction should be enforced on the API end and is subject to change. Because of that, I ripped it out.

Reply:
ACTUALLY, `manager_email` is no longer present anymore, at all. So it was a good thing to remove this. I have also pushed a commit to remove that variable altogether. cc @rocktavious and @davidbloss for visibility.

Reply:
fwiw this still works in CLI.

```
$ opslevel get team complex_tf_team | jq '.Manager'
{
  "Id": "Z2lkOi8vb3BzbGV2ZWwvVXNlci8xNDYwMA",
  "Email": "kyle+orange@opslevel.com",
  "HTMLUrl": "https://app.opslevel.com/users/14600",
  "Name": "Kyle",
  "Role": "admin"
}
```

It's autodetected, has nothing to do with `manager_email`

Reply:
Its probably not removed entirely but deprecated